application/data/tables.py

```diff
@@ -0,0 +1,295 @@
+
+        elif TableObjectDataBuilder.__is_ethnicity_column(table_settings["tableOptions"]["table_category_column"]):
+            return TableObjectDataBuilder.__get_v2_ethnicity_is_rows_data(
+                table_object, table_settings, dictionary_lookup
+            )
+
+        elif TableObjectDataBuilder.__is_ethnicity_column(table_settings["tableOptions"]["table_group_column"]):
+            return TableObjectDataBuilder.__get_v2_ethnicity_is_columns_data(
+                table_object, table_settings, dictionary_lookup
+            )
+
+        return {}
+
+    @staticmethod
+    def __get_v2_simple_data(table_object, table_settings, dictionary_lookup):
+        table_data_object = TableObjectDataBuilder.build(table_object)
+        table_columns = TableObjectDataBuilder.__get_table_columns(table_settings)
+        headers = ["Ethnicity"] + table_columns
+        return [headers] + table_data_object["data"][1:]
+
+    @staticmethod
+    def __get_v2_ethnicity_is_rows_data(table_object, table_settings, dictionary_lookup):
+        data = TableObjectDataBuilder.__get_harmonised_data(table_settings, dictionary_lookup)
+
+        required_columns = [
+            table_settings["tableOptions"]["table_category_column"],
+            table_settings["tableOptions"]["table_group_column"],
+        ] + TableObjectDataBuilder.__get_table_columns(table_settings)
+        if table_settings["tableOptions"]["table_column_order_column"] not in required_columns:
+            required_columns.append(table_settings["tableOptions"]["table_column_order_column"])
+
+        headers = data[0]
+        required_indices = [
+            headers.index(required_column) for required_column in required_columns if required_column in headers
+        ]
+
+        converted = [[row[index] for index in required_indices] for row in data]
+        converted[0][0] = "Ethnicity"
+
+        return converted
+
+    @staticmethod
+    def __get_v2_ethnicity_is_columns_data(table_object, table_settings, dictionary_lookup):
+
+        data = TableObjectDataBuilder.__get_harmonised_data(table_settings, dictionary_lookup)
+
+        required_columns = [
+            table_settings["tableOptions"]["table_group_column"],
+            table_settings["tableOptions"]["table_category_column"],
+        ] + TableObjectDataBuilder.__get_table_columns(table_settings)
+        if table_settings["tableOptions"]["table_order_column"] not in required_columns:
+            required_columns.append(table_settings["tableOptions"]["table_order_column"])
+
+        headers = data[0]
+        required_indices = [
+            headers.index(required_column) for required_column in required_columns if required_column in headers
+        ]
+
+        converted = [[row[index] for index in required_indices] for row in data]
+        converted[0][0] = "Ethnicity"
+
+        return converted
+
+    @staticmethod
+    def __get_harmonised_data(table_settings, dictionary_lookup):
```

Comment:
Does this need a new name to eliminate harmony?

Reply:
yep